### PR TITLE
BOOL to bool, pt. 2

### PR DESCRIPTION
**Pull request by larsnaesbye**
With over twice as many substitutions as the previous patch (a trend I hope to continue), here's another bunch of conversions from the int-based booleans to true booleans.

RW Testing has consisted of compiling, executing the app on macOS 14.3 on Apple Silicon running in Rosetta and 'doing a few moves'.

### CODE REVIEW

File: indra/llaudio/llaudiodecodemgr.cpp [hunk 1]
```diff
@@ -70,22 +70,22 @@ class LLVorbisDecodeState : public LLThreadSafeRefCount
...
+	bool decodeSection(); // Return true if done.
+	bool finishDecode();
 
 	void flushBadFile();
 
 	void ioComplete(S32 bytes)			{ mBytesRead = bytes; }
-	BOOL isValid() const				{ return mValid; }
-	BOOL isDone() const					{ return mDone; }
+	bool isValid() const				{ return mValid; }
+	bool isDone() const					{ return mDone; }
```
[the remark this answers is not in the document]
Comment (Ansariel) [reply indra/llaudio/llaudiodecodemgr.cpp:81]: Looks good to me

**Comment by Ansariel** [comment]
Looks good to me

File: indra/llcommon/is_approx_equal_fraction.h [hunk 2]
```diff
@@ -45,7 +45,7 @@
 template <typename FTYPE>
 inline BOOL is_approx_equal_fraction_impl(FTYPE x, FTYPE y, U32 frac_bits)
```
Comment (Ansariel) [inline indra/llcommon/is_approx_equal_fraction.h:46]: You forgot the return type and the type for "ret" two lines below.

**Comment by Ansariel** [comment]
I think you forgot to change the return type in some cases.

File: indra/llcommon/is_approx_equal_fraction.h (hunk 2 above)
Comment (larsnaesbye) [reply indra/llcommon/is_approx_equal_fraction.h:46]: Where is that file included from? It is nowhere to be found in the Xcode project....

File: indra/llappearance/llavatarjoint.h [hunk 3]
```diff
@@ -62,7 +62,7 @@ class LLAvatarJoint :
 	virtual BOOL isTransparent() { return mIsTransparent; }
 
 	// Returns true if this object should inherit scale modifiers from its immediate parent
-	virtual BOOL inheritScale() { return FALSE; }
+	virtual BOOL inheritScale() { return false; }
```
Comment (Ansariel) [inline indra/llappearance/llavatarjoint.h:65]: Return type

File: indra/llappearance/llavatarjoint.h [hunk 4]
```diff
@@ -127,7 +127,7 @@ class LLAvatarJointCollisionVolume : public LLAvatarJoint
 	LLAvatarJointCollisionVolume();
 	virtual ~LLAvatarJointCollisionVolume() {};
 
-	/*virtual*/ BOOL inheritScale() { return TRUE; }
+	/*virtual*/ BOOL inheritScale() { return true; }
```
Comment (Ansariel) [inline indra/llappearance/llavatarjoint.h:130]: Return type

File: indra/llappearance/llavatarjointmesh.cpp [hunk 5]
```diff
@@ -186,7 +186,7 @@ BOOL LLAvatarJointMesh::allocateSkinData( U32 numSkinJoints )
 {
 	mSkinJoints = new LLSkinJoint[ numSkinJoints ];
 	mNumSkinJoints = numSkinJoints;
-	return TRUE;
+	return true;
```
Comment (Ansariel) [inline indra/llappearance/llavatarjointmesh.cpp:189]: Return type

File: indra/llappearance/llavatarjointmesh.cpp [hunk 6]
```diff
@@ -119,7 +119,7 @@ BOOL LLSkinJoint::setupSkinJoint( LLAvatarJoint *joint)
 	mRootToParentJointSkinOffset = totalSkinOffset(getBaseSkeletonAncestor(joint));
 	mRootToParentJointSkinOffset = -mRootToParentJointSkinOffset;
 
-	return TRUE;
+	return true;
```
Comment (Ansariel) [inline indra/llappearance/llavatarjointmesh.cpp:122]: Return type

**Comment by Ansariel** [comment]
Those are just a few as I didn't want to spam you with all the return types stuff.

**Comment by larsnaesbye** [comment]
> Those are just a few as I didn't want to spam you with all the return types stuff.

Thank you. I am doing a lot of search-and-replace and then pushing so maybe I should make this a draft PR for now?

**Comment by larsnaesbye** [comment]
I can't convert it to a draft, apparently.

File: indra/llappearance/llavatarjoint.cpp [hunk 7]
```diff
@@ -167,10 +167,10 @@ void LLAvatarJoint::updateJointGeometry()
 }
 
 
-BOOL LLAvatarJoint::updateLOD(F32 pixel_area, BOOL activate)
+bool LLAvatarJoint::updateLOD(F32 pixel_area, bool activate)
 {
-	BOOL lod_changed = FALSE;
-	BOOL found_lod = FALSE;
+	bool lod_changed = FALSE;
```
Comment (Ansariel) [inline indra/llappearance/llavatarjoint.cpp:172]: 🤔 😄

Comment (larsnaesbye) [reply indra/llappearance/llavatarjoint.cpp:172]: Yeah, that needs to get back in. Thank you for finding it :-D

**Review by marchcat — APPROVED** [review]
Thank you!

Visual Studio found couple missed values, I'll fix them within the next commit.

**Pull request merged**